system/Database/Query.php

```diff
@@ -275,11 +275,13 @@ protected function compileBinds()
     {
         $sql = $this->finalQueryString;
 
-        $hasNamedBinds = preg_match('/:((?!=).+):/', $sql) === 1;
+        $hasBinds      = strpos($sql, $this->bindMarker) !== false;
+        $hasNamedBinds = ! $hasBinds
+            && preg_match('/:(?!=).+:/', $sql) === 1;
 
         if (empty($this->binds)
             || empty($this->bindMarker)
-            || (! $hasNamedBinds && strpos($sql, $this->bindMarker) === false)
+            || (! $hasNamedBinds && ! $hasBinds)
```

Comment:
$hasBinds is essentially duplicated here. Instead, we can either:
1. Interchange the operands, with `strpos` being called first
2. Retain $hasNamedBinds only, as it contains already the $hasBinds check

Reply:
@paulbalandan 
1
I don't understand your intention. I removed duplicated `strpos()` call.

2
If I remove `&& ! $hasBinds`, it breaks.
https://3v4l.org/UWTvF

Reply:
Okay. I stand corrected with my second suggestion. Basically, I want something like this.
https://3v4l.org/V4pJv